README.md

```diff
@@ -1,62 +1,48 @@
-5. Do not allow bypassing the above settings.
-
-#### Actions/General
-1. Allow all actions and reusable workflows.
-2. Allow GitHub Actions to create and approve pull requests (if dependabot is used and it's configured to automatically
-merge pull requests).
-
-### Specification
-The `spec.go` file provides a programmatic representation of the configuration options.
-
-### Configuration
-This template provides two types of "configs":
-* general configuration (that applies to both, sources and destinations)
-* and source/destination specific configs.
-
-All are defined in `config.go` and are represented by separate types. General configs should be added to `pinecone.Config`,
-whereas any source or destination specific configs should be added to `pinecone.SourceConfig` and 
-`pinecone.DestinationConfig` respectively.
+# Conduit destination connector for Pinecone
+
+The Pinecone connector is one of [Conduit](https://github.com/ConduitIO/conduit) standalone plugins. It provides a destination connector for [Pinecone](https://www.Pinecone.io/).
+
+It uses the [gRPC go Pinecone client](github.com/Pinecone-io/go-Pinecone) to connect to Pinecone.
+
+## Which OpenCDC data format does the destination connector accept?
+
+The destination connector expects an OpenCDC formatted record with the following fields
+
+| Field                   | Description                                                                             |
+|-------------------------|-----------------------------------------------------------------------------------------|
+| `record.Operation`      | which conduit operation does the record do.                                             |
+| `record.Metadata`       | a string to string map representing the Pinecone vector metadata.                       |
+| `record.Key`            | the vector id that is being updated.                                                    |
+| `record.Payload.Before` | <discarded>                                                                                 |
+| `record.Payload.After`  | the vector body, in json format             | 
+| `(json) record.Payload.After.values`  | an array of float32 representing the vector values              | 
+| `(json) record.Payload.After.sparse_values`  | **(optional)** the sparse vector values               | 
+| `(json) record.Payload.After.sparse_values.indices`  | an array of uint32 representing the sparse vector indices              | 
+| `(json) record.Payload.After.sparse_values.values`  | an array of float32 representing the sparse vector values               | 
+
+## How to Build?
+
+Run `make build` to compile the connector.
+
+## Testing
+
+To perform the tests locally you'll need the `API_KEY` and `HOST_URL` environment variables set. To do so:
+
+1. You'll need to setup a new account if you don't have it at https://www.pinecone.io/   
+2. Create a new index with `cosine` as the metric, and copy the host url.
```

Comment:
Is there a special reason to use `cosine` as the metric? What would happen if we used a different metric?

Reply:
I tried to run tests with all the index metric types, and they run perfectly. They must be set to 2 dimensions though.